python_digest/__init__.py

```diff
@@ -113,7 +110,7 @@ def calculate_nonce(timestamp, secret, salt=None):
     if not salt:
         salt = ''.join([random.choice('0123456789ABCDEF') for x in range(4)])
     return b"%s:%s:%s" % (timestamp, salt,
```

Comment:
I think this should be a unicode literal, not a bytes literal, but I'm going to handle that in a separate PR just to isolate tests.